### DISCUSSION TITLE
Tagging multi-part uploads so the tags can be used in S3 Metric filters

### DISCUSSION THREAD

**Discussion by okachobi** [aws/aws-sdk-js#4184] [Q&A]
I found issue [1412](https://github.com/aws/aws-sdk-js/issues/1412) which appears to be closed with tagging AFTER the data has been transmitted.

Is it not possible to tag multi-part uploads?  Within the S3 console there are request Metrics which can be filtered on prefix, endpoint arn, or tag.  Today I needed to look at the "Uploaded bytes" metrics filtered on a specific tag that we pass to the Upload() as an option.  I found that there was virtually no "uploaded bytes" for the tag I was filtering on and I knew this customer was uploading a lot of data.  I opened a ticket with AWS support and started search to find the above issue and I believe the fact that the data is tagged after it is uploaded prevents it from counting against the uploaded bytes for the filtered tag.

Is it impossible to tag the multi-parts so that they count against your tag-filtered metrics?

**Comment by ajredniwja** [comment]
Hey @okachobi thanks for opening this issue, apologies it got longer, can you please share your support case number/id, from the PR I see https://github.com/aws/aws-sdk-js/pull/1425 you can tag multi-parts

**Reply by okachobi** [reply]
The support case number was 10580065901 : "Bucket Metrics -> Request metrics appear to not reflect multipart puts"
